pkg/openshift/util/console_links.go

```diff
@@ -0,0 +1,62 @@
+package util
+
+import (
+	"encoding/json"
+	"errors"
+
+	hawtiov1alpha1 "github.com/hawtio/hawtio-operator/pkg/apis/hawtio/v1alpha1"
+
+	consolev1 "github.com/openshift/api/console/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// GetHawtconfig reads the console configuration from the config map
+func GetHawtconfig(configMap *corev1.ConfigMap) (*hawtiov1alpha1.Hawtconfig, error) {
+	var hawtconfig *hawtiov1alpha1.Hawtconfig
+
+	hawtconfigJSON, ok := configMap.Data["hawtconfig.json"]
+	if !ok {
+		return hawtconfig, errors.New("Did not find hawtconfig.json in ConfigMap")
+	}
+	err := json.Unmarshal([]byte(hawtconfigJSON), &hawtconfig)
+	if err != nil {
+		return hawtconfig, err
+	}
+
+	return hawtconfig, nil
+}
+
+// NewConsoleLink creates a ConsoleLink instance
+func NewConsoleLink(name string, route *routev1.Route, hawtconfig *hawtiov1alpha1.Hawtconfig) *consolev1.ConsoleLink {
+	consoleLink := &consolev1.ConsoleLink{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   name,
+			Labels: map[string]string{"app": "hawtio"},
+		},
+		Spec: consolev1.ConsoleLinkSpec{
+			Link: consolev1.Link{
+				Text: "",
+				Href: "",
+			},
+			Location: consolev1.ApplicationMenu,
+			ApplicationMenu: &consolev1.ApplicationMenuSpec{
+				Section:  "",
+				ImageURL: "",
+			},
+		},
```

Comment:
The struct is already defaulted with empty strings so that can be removed.

Reply:
Done. Note that `Location` needs to be set to `ApplicationMenu` and `ApplicationMenu` is nil by default.